### PR TITLE
ToCode Solution: תרגול פונקציות

### PR DESCRIPTION
**Pull request by HananRosenthal**

### CODE REVIEW

File: 14_functions_lab/05.py [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-            myDict[c] = list
+    for word in words:
+        k = f(word)
+        list = [w for w in words if w[0] == k]
+        myDict[k] = list
     print(myDict)
 
-words = ["hello", "hi", "help", "bye", "here", "modify", "movie", "ticho", "time"]
+words = ["hello", "hi", "help", "bye", "here", "modify", "movie", "typo", "time"]
 groupby(f, *words)
```
Comment (szabgab) [inline 14_functions_lab/05.py:12]: This code still goes over the input O(n^2) once in the for loop and once in the list comprehension. You don't need the latter.

Comment (HananRosenthal) [inline 14_functions_lab/05.py:12]: the later is required to compose a list of words begining with a certain letter

**Comment by szabgab** [comment]
You already have the key in hand when you call `k = f(word)`. Use that as the key in the dict.

**Comment by HananRosenthal** [comment]
yes, but still need to extract all words belonging to a specific key. How would you do that without another list comprehension?

**Comment by szabgab** [comment]
You go over the words one by one and on each turn you add the word to the list of the respective key.
There is only one "loop" in this sentence.

File: 14_functions_lab/05.py [hunk 2]
```diff
@@ -1,15 +1,12 @@
...
     for word in words:
         k = f(word)
-        list = [w for w in words if w[0] == k]
-        myDict[k] = list
+        myDict.setdefault(k, []).append(word)
     print(myDict)
 
-words = ["hello", "hi", "help", "bye", "here", "modify", "movie", "typo", "time"]
+words = ["hello", "hi", "help", "bye", "here", "modify", "movie", "typo", "time", 'jump', 'jumbo', 'murder', 'jamboree']
 groupby(f, *words)
```
Comment (szabgab) [inline 14_functions_lab/05.py:12]: 👍 yes, that's it!

Comment (HananRosenthal) [inline 14_functions_lab/05.py:12]: sweated blood...

Comment (szabgab) [inline 14_functions_lab/05.py:12]: Have a 🍪  you deserve it :)

**Pull request closed without merging**